### PR TITLE
feat: add ability to exclude paths for stories (#285)

### LINKED ISSUE #285: [6.0] option to exclude paths from get stories and the watcher
**Is your feature request related to a problem? Please describe.**

It can be slow in large repos to run the get stories or watcher command and having an exclude option could be useful for avoiding that.

**Describe the solution you'd like**

Something like this would be useful and wouldn't break compatibility with the config file used in other frameworks

```
reactNativeOptions:{
  excludePaths: []
} 
```

**Are you able to assist bring the feature to reality?**
Yes

### PR DESCRIPTION
**Pull request by jgornick**
Adds ability to specify excluded paths in main.js configuration when searching for stories. This helps with glob performance by ignoring node_modules by default.

Fixes #285

### CODE REVIEW

File: app/react-native/scripts/loader.js [hunk 1]
```diff
@@ -36,8 +36,18 @@ function writeRequires({ configPath, absolute = false }) {
   const storybookRequiresLocation = path.resolve(cwd, configPath, 'storybook.requires.js');
 
   const main = getMain({ configPath });
+  const reactNativeOptions = main.reactNativeOptions;
+  const excludePaths = reactNativeOptions && reactNativeOptions.excludePaths
```
Comment (dannyhw) [inline app/react-native/scripts/loader.js]: I think it would be a good idea to split this up a bit, I think 2 ternaries in one assignment is a bit hard to follow.

**Comment by dannyhw** [comment]
@jgornick Thanks for your contribution 🙇 .

Its looking great, if you can make the one change I suggested then I think this is good to go 👍

File: app/react-native/scripts/loader.js (hunk 1 above)
Comment (jgornick) [reply app/react-native/scripts/loader.js]: Addressed :)

**Comment by dannyhw** [comment]
Great work!
